doc/contributing.rst

```diff
@@ -3,15 +3,140 @@ Contributor Guide
 
 This package is in very early stages. Lots of work is needed.
 
-GitHub repo: `https://github.com/xgcm/xhistogram <https://github.com/xgcm/xhistogram>`_
+You can help out just by using ``xhistogram`` and reporting
+`issues <https://github.com/xgcm/xhistogram/issues>`__.
+
+The following sections cover some general guidelines for maintainers and
+contributors wanting to help develop ``xhistogram``.
+
+
+Feature requests, suggestions and bug reports
+---------------------------------------------
+
+We are eager to hear about any bugs you have found, new features you
+would like to see and any other suggestions you may have. Please feel
+free to submit these as `issues <https://github.com/xgcm/xhistogram/issues>`__.
+
+When suggesting features, please make sure to explain in detail how
+the proposed feature should work and to keep the scope as narrow as
+possible. This makes features easier to implement in small PRs.
+
+When report bugs, please include:
+
+* Any details about your local setup that might be helpful in
+  troubleshooting, specifically the Python interpreter version, installed
+  libraries, and ``xhistogram`` version.
+* Detailed steps to reproduce the bug, ideally a `Minimal, Complete and
+  Verifiable Example <http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports>`__.
+* If possible, a demonstration test that currently fails but should pass
+  when the bug is fixed.
+
+
+Write documentation
+-------------------
+Adding documentation is always helpful. This may include:
+
+* More complementary documentation. Have you perhaps found something unclear?
+* Docstrings.
+* Example notebooks of ``xhistogram`` being used in real analyses.
+
+The ``xhistogram`` documentation is written in reStructuredText. You
+can follow the conventions in already written documents. Some helpful guides
+can be found
+`here <http://docutils.sourceforge.net/docs/user/rst/quickref.html>`__ and
+`here <https://github.com/ralsina/rst-cheatsheet/blob/master/rst-cheatsheet.rst>`__.
+
+When writing and editing documentation, it can be useful to see the resulting
+build without having to push to Github. You can build the documentation locally
+by running::
+
+    $ # Install the packages required to build the docs in a conda environment
+    $ conda env update -f doc/environment.yml
+    $ conda activate xhistogram_doc_env
```

Comment:
We _may_ want to use `conda env create` here instead of `conda env update`. For example, if someone is updating an existing conda environment not named `xhistogram_doc_env` then `conda activate xhistogram_doc_env` may result in an error.